git_theta/utils.py

```diff
@@ -61,3 +62,33 @@ def unflatten(d: Dict[Tuple[str], Any]) -> Dict[str, Union[Dict[str, Any], Any]]
             curr = curr.setdefault(k, {})
         curr[ks[-1]] = v
     return nested
+
+
+def is_valid_oid(oid: str) -> bool:
```

Comment:
Is there somewhere where we canonicalize hex hash values to be lower case? There are a few lower case assumptions sprinkled around the code base. 

Only working with lowercase is fine as long as we can enforce it if any tool we integrate with decides to switch to upper. Might make sense to add this as an issue instead of fixing it in this PR